serialization/php_serialized_value.ksy

```diff
@@ -0,0 +1,260 @@
+      - id: value_dec
+        type: str
+        terminator: 0x3b # ';'
+        doc: The value of the `int`, in ASCII decimal.
+    instances:
+      value:
+        value: value_dec.to_i
+        doc: The value of the `int`, parsed as an integer.
+    doc: |
+      The contents of an integer-like value: either an actual integer (`value_type::int`) or a reference (`value_type::variable_reference`, `value_type::object_reference`).
+  float_contents:
+    seq:
+      - id: colon
+        contents: ':'
+      - id: value_dec
+        type: str
+        terminator: 0x3b # ';'
+        doc: |
+          The value of the `float`, in ASCII decimal, as generated by PHP's usual double-to-string conversion. In particular, this means that:
+          
+          * A decimal point may not be included (for integral numbers)
+          * The number may use exponent notation (e. g. `1.0E+16`)
+          * Positive and negative infinity are represented as `INF` and `-INF`, respectively
+          * Not-a-number is represented as `NAN`
+    doc: The contents of a floating-point value.
+  size_prefixed_quoted_string:
+    seq:
+      - id: size_dec
+        type: str
+        terminator: 0x3a # ':'
+        doc: The size of the string's data in bytes, in ASCII decimal. The quotes are not counted in this size number.
+      - id: opening_quote
+        contents: '"'
+      - id: data
+        size: size
+        doc: The data contained in the string. The quotes are not included.
+      - id: closing_quote
+        contents: '"'
+    instances:
+      size:
+        value: size_dec.to_i
+        doc: The size of the string's contents in bytes, parsed as an integer. The quotes are not counted in this size number.
+    doc: |
+      A quoted string prefixed with its size.
+      
+      Despite the quotes surrounding the string data, it can contain arbitrary bytes, which are never escaped in any way. This does not cause any ambiguities when parsing - the bounds of the string are determined only by the size field, not by the quotes.
+  string_contents:
+    seq:
+      - id: colon
+        contents: ':'
+      - id: string
+        type: size_prefixed_quoted_string
+      - id: semicolon
+        contents: ';'
+    instances:
+      value:
+        value: string.data
+        doc: The value of the string, as a byte array.
+    doc: |
+      The contents of a string value.
+      
+      Note: PHP strings can contain arbitrary byte sequences. They are not necessarily valid text in any specific encoding.
+  mapping_entry:
+    seq:
+      - id: key
+        type: php_serialized_value
+        doc: The key of the entry.
+      - id: value
+        type: php_serialized_value
+        doc: The value of the entry.
+    doc: A mapping entry consisting of a key and a value.
+  size_prefixed_mapping:
+    seq:
+      - id: size_dec
+        type: str
+        terminator: 0x3a # ':'
+        doc: The number of key-value pairs in the mapping, in ASCII decimal.
```

Comment:
It's not just some "size", it's "number of entries", thus please consider using `num_entries_dec` and `num_entries` for clarity.

Reply:
Good point, "size" isn't the best term here. (I'm guessing this was a copy-paste artifact.)